From 15d411a1b7409bff097b4c9381e59abd3b124907 Mon Sep 17 00:00:00 2001
From: lovelylain <jobmailcn@gmail.com>
Date: Sun, 20 Nov 2022 11:43:10 +0800
Subject: [PATCH] patch dockerd.init

---
 utils/dockerd/files/dockerd.init | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/utils/dockerd/files/dockerd.init b/utils/dockerd/files/dockerd.init
index fc3b88a6..ef35e31e 100755
--- a/utils/dockerd/files/dockerd.init
+++ b/utils/dockerd/files/dockerd.init
@@ -17,6 +17,16 @@ json_add_array_string() {
 	json_add_string "" "${1}"
 }
 
+json_merge_object() {
+	eval "`jshn -r "$1"|tail +2`"
+}
+
+json_add_object_string() {
+	json_add_object "$1"
+	json_merge_object "$2"
+	json_close_object
+}
+
 find_network_device() {
 	local device="${1}"
 	local device_section=""
@@ -40,7 +50,9 @@ find_network_device() {
 }
 
 boot() {
-	uciadd
+	config_load dockerd
+	config_get_bool iptables globals iptables "1"
+	[ "${iptables}" -eq "1" ] && uciadd
 	rc_procd start_service
 }
 
@@ -151,6 +163,7 @@ ucidel() {
 
 process_config() {
 	local alt_config_file data_root log_level iptables bip
+	local log_driver log_opts
 
 	[ -f /etc/config/dockerd ] || {
 		# Use the daemon default configuration
@@ -175,6 +188,9 @@ process_config() {
 
 	# Don't add these options by default
 	# omission == docker defaults
+	config_get http_proxy globals http_proxy ""
+	config_get log_driver globals log_driver ""
+	config_get log_opts globals log_opts ""
 	config_get bip globals bip ""
 	config_get registry_mirrors globals registry_mirrors ""
 	config_get hosts globals hosts ""
@@ -189,6 +205,8 @@ process_config() {
 	json_add_string "data-root" "${data_root}"
 	json_add_string "log-level" "${log_level}"
 	json_add_boolean "iptables" "${iptables}"
+	[ -z "${log_driver}" ] || json_add_string "log-driver" "${log_driver}"
+	[ -z "${log_opts}" ] || json_add_object_string "log-opts" "${log_opts}"
 	[ -z "${bip}" ] || json_add_string "bip" "${bip}"
 	[ -z "${registry_mirrors}" ] || json_add_array "registry-mirrors"
 	[ -z "${registry_mirrors}" ] || config_list_foreach globals registry_mirrors json_add_array_string
@@ -215,6 +233,7 @@ start_service() {
 
 	procd_open_instance
 	procd_set_param stderr 1
+	[ -z "${http_proxy}" ] || procd_set_param env "HTTP_PROXY=${http_proxy}"
 	if [ -z "${DOCKERD_CONF}" ]; then
 		procd_set_param command /usr/bin/dockerd
 	else
-- 
2.25.1

